src/Screens/Explore/DiveSite.tsx

```diff
@@ -153,18 +150,16 @@ const DiveSite: FunctionComponent<DiveSiteProps> = ({ navigation, route }) => {
       dispatch(handleFetchReviews(currentSpotId));
     }
 
-    // because we need to fetch images as well, we're making calls to the api unless the
-    // site we  want has had its images downloaded
-    if (!diveSiteInState || (diveSiteInState && !diveSite.images)) {
-      dispatch(handleFetchDiveSite(currentSpotId));
-    }
-  }, [currentSpotId, dispatch, reviewInState, diveSiteInState, diveSite]);
+    // just make call to fetch beach if no images, since we're only calling this hook once now
+    dispatch(handleFetchDiveSite(currentSpotId));
+  }, [currentSpotId]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
hmm the old dependencies seem more correct imo. although we could probably avoid fetching the nearby sites too frequently. maybe all these fetches should all be in different effects

Reply:
So I removed it because I figured that function only needs to run once, when the currentSpotId changes. All of the other dependencies will only trigger re-renders. None of those calls need to happen more than once.

Reply:
the correct dependency array is `[currentSpotId, dispatch, diveSiteInState, diveSite]` so you can remove reviewInState. I think otherwise the correct change would be to split these up into separate effects. then you can delete the linter comment, and still only have it run the correct number of times for each fetch